src/mono/netcore/System.Private.CoreLib/src/System/Threading/ThreadPool.Browser.Mono.cs

```diff
@@ -10,26 +10,49 @@
 
 namespace System.Threading
 {
-    public sealed class RegisteredWaitHandle : MarshalByRefObject
+    public sealed partial class Thread
     {
-        internal RegisteredWaitHandle(WaitHandle waitHandle, _ThreadPoolWaitOrTimerCallback callbackHelper,
-            int millisecondsTimeout, bool repeating)
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal void ResetThreadPoolThread()
         {
+            Debug.Assert(this == CurrentThread);
+            Debug.Assert(IsThreadPoolThread);
+
+            if (_mayNeedResetForThreadPool)
+            {
+                ResetThreadPoolThreadSlow();
+            }
         }
+    }
 
+    public sealed partial class RegisteredWaitHandle : MarshalByRefObject
+    {
         public bool Unregister(WaitHandle? waitObject)
         {
             throw new PlatformNotSupportedException();
         }
     }
 
+    internal sealed partial class CompleteWaitThreadPoolWorkItem : IThreadPoolWorkItem
+    {
+        void IThreadPoolWorkItem.Execute()
+        {
+            Debug.Fail("Registered wait handles are currently not supported");
+        }
+    }
+
     public static partial class ThreadPool
     {
+        internal const bool SupportsTimeSensitiveWorkItems = true;
```

Comment:
Can you add some comment about what this does

Reply:
Done, also changed to false to work similarly to before and avoid creating/checking the extra queue for now